action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -406,6 +412,17 @@ func (stateDB *StateDBAdapter) Suicide(evmAddr common.Address) bool {
 		stateDB.logError(err)
 		return false
 	}
+	// before calling Suicide, EVM will transfer the contract's balance to beneficiary
+	// need to create a transaction log on successful suicide
+	if len(stateDB.lastAddBalanceAmount.Bytes()) > 0 {
```

Comment:
should be check `if stateDB.lastAddBalanceAmount != nil && stateDB.lastAddBalanceAmount.Cmp(big.NewInt(0))>0` to prevent nil or negative amount

Reply:
in `AddBalance()`, nil or negative amount will be caught and return error, in other words, `lastAddBalanceAmount` must be valid positive number here

add initialization of the value `lastAddBalanceAmount` in `new` to prevent being nil